.github/scripts/compile.sh

```diff
@@ -33,30 +38,45 @@ do_patch () {
     fi
 }
 
-WORKING=`pwd`
-wget ${PROTON_SOURCE_URL} -O qpid-proton.tar.gz
+do_patch "patches/proton" qpid-proton-src
 
-mkdir -p qpid-proton-src build staging proton_build proton_install
-tar -zxf qpid-proton.tar.gz -C qpid-proton-src --strip-components 1
+do_build () {
+  local suffix=${1}
+  local runtime_check=${2}
 
-do_patch "patches/proton" qpid-proton-src
+  cmake -S "$WORKING/qpid-proton-src" -B "$WORKING/proton_build${suffix}" \
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+    -DRUNTIME_CHECK="${runtime_check}" \
+    -DENABLE_LINKTIME_OPTIMIZATION=ON \
+    -DCMAKE_POLICY_DEFAULT_CMP0069=NEW -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
+    -DBUILD_TLS=ON -DSSL_IMPL=openssl -DBUILD_STATIC_LIBS=ON -DBUILD_BINDINGS=python -DSYSINSTALL_PYTHON=ON \
+    -DBUILD_EXAMPLES=OFF -DBUILD_TESTING=OFF \
+    -DCMAKE_INSTALL_PREFIX=/usr
+  cmake --build "proton_build${suffix}" --parallel "$(nproc)" --verbose
+  DESTDIR="$WORKING/proton_install${suffix}" cmake --install "proton_build${suffix}"
+
+  cmake -S "$WORKING/" -B "$WORKING/build${suffix}" \
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+    -DRUNTIME_CHECK="${runtime_check}" \
+    -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
+    -DProton_USE_STATIC_LIBS=ON -DUSE_LIBWEBSOCKETS=ON -DUSE_LIBNGHTTP2=ON \
+    -DProton_DIR="$WORKING/proton_install${suffix}/usr/lib64/cmake/Proton" \
+    -DBUILD_TESTING=OFF \
+    -DVERSION="${VERSION}" \
+    -DCMAKE_INSTALL_PREFIX=/usr
+  cmake --build "$WORKING/build${suffix}" --parallel "$(nproc)" --verbose
+}
+
+do_build "" OFF
+do_build "_asan" asan
+do_build "_tsan" tsan
```

Comment:
The image build used to take 6 mins on GHA, this PR ran 14 minutes. Too bad. Each of these builds is different (different gcc options) so there is no space for any caching or such ways.